### ISSUE TITLE
Steam client segfault during first run

### ISSUE THREAD

**Issue by japhir** [ValveSoftware/steam-for-linux#5184]
#### Your system information

* Steam client version (build number or date):  1.0.0.54 on Arch Linux Distribution (e.g. Ubuntu): 
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)

Client segfaults when I run it for the first time. I've never had the steam client work on my Arch linux machine, and I've dedicated quite some time to trying it. It's a shame that I now still have to dualboot to Windows.

steam-native
```
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
/home/japhir/.local/share/Steam/steam.sh: line 99: no match: **:
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
Running Steam on arch  64-bit
STEAM_RUNTIME is disabled by the user
Installing breakpad exception handler for appid(steam)/version(0)
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[2017-09-25 00:08:10] Startup - updater built Nov 23 2016 01:05:42
crash_20170925000810_1.dmp[1129]: Uploading dump (out-of-process)
/tmp/dumps/crash_20170925000810_1.dmp
/home/japhir/.local/share/Steam/steam.sh: line 713:  1126 Aborted                 $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
/home/japhir/.local/share/Steam/steam.sh: line 444: no match: ssfn*
crash_20170925000810_1.dmp[1129]: Finished uploading minidump (out-of-process): success = yes
crash_20170925000810_1.dmp[1129]: response: CrashID=bp-9c1b18f4-9378-4ecf-88ff-a6c8e2170924
crash_20170925000810_1.dmp[1129]: file ''/tmp/dumps/crash_20170925000810_1.dmp'', upload yes: ''CrashID=bp-9c1b18f4-9378-4ecf-88ff-a6c8e2170924''
```
after `rm -rf .steam .local/share/Steam` I run the other version:
steam-runtime
```
Setting up Steam content in /home/japhir/.local/share/Steam
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
/home/japhir/.local/share/Steam/steam.sh: line 99: no match: **:
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
Running Steam on arch  64-bit
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
STEAM_RUNTIME is enabled automatically
Installing breakpad exception handler for appid(steam)/version(0)
*** Error in `/home/japhir/.local/share/Steam/ubuntu12_32/steam': free(): invalid pointer: 0xf77a3810 ***
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[2017-09-25 00:14:31] Startup - updater built Nov 23 2016 01:05:42
crash_20170925001431_1.dmp[2257]: Uploading dump (out-of-process)
/tmp/dumps/crash_20170925001431_1.dmp
/home/japhir/.local/share/Steam/steam.sh: line 713:  2254 Aborted                 $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
/home/japhir/.local/share/Steam/steam.sh: line 444: no match: ssfn*
crash_20170925001431_1.dmp[2257]: Finished uploading minidump (out-of-process): success = yes
crash_20170925001431_1.dmp[2257]: response: CrashID=bp-4879bee9-4cd5-4036-ac5f-15a6a2170924
crash_20170925001431_1.dmp[2257]: file ''/tmp/dumps/crash_20170925001431_1.dmp'', upload yes: ''CrashID=bp-4879bee9-4cd5-4036-ac5f-15a6a2170924''
```

#### Steps for reproducing this issue:

1. Install steam
2. Install lib32-nvidia driver for my ancient video card (NVIDIA GeForce 9800 GT)
3. Install lib32-nvidia utils and libgl
4. Install arch steam-native-runtime package, containing a lot of the libraries needed for steam and ttf-liberation, an Arial font replacement that should work
5. Reboot
6. Start steam-native or steam-runtime
7. Click accept on the terms dialog
-- the app crashes.

**Comment by kisak-valve** [comment]
Hello @japhir, can you try to run `LC_ALL=C steam` in a terminal?

**Comment by japhir** [comment]
LC_ALL=C steam gives me
```
Setting up Steam content in /home/japhir/.local/share/Steam
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
/home/japhir/.local/share/Steam/steam.sh: line 99: no match: **:
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
Running Steam on arch  64-bit
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
STEAM_RUNTIME is enabled automatically
Installing breakpad exception handler for appid(steam)/version(0)
*** Error in `/home/japhir/.local/share/Steam/ubuntu12_32/steam': free(): invalid pointer: 0xf7779810 ***
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[2017-09-26 07:46:27] Startup - updater built Nov 23 2016 01:05:42
crash_20170926074627_1.dmp[1929]: Uploading dump (out-of-process)
/tmp/dumps/crash_20170926074627_1.dmp
/home/japhir/.local/share/Steam/steam.sh: line 713:  1926 Aborted                 $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
/home/japhir/.local/share/Steam/steam.sh: line 444: no match: ssfn*
crash_20170926074627_1.dmp[1929]: Finished uploading minidump (out-of-process): success = yes
crash_20170926074627_1.dmp[1929]: response: CrashID=bp-0d00c7b8-cbb1-4142-a99c-499bb2170925
crash_20170926074627_1.dmp[1929]: file ''/tmp/dumps/crash_20170926074627_1.dmp'', upload yes: ''CrashID=bp-0d00c7b8-cbb1-4142-a99c-499bb2170925''
```

**Comment by japhir** [comment]
> LC_ALL=C steam-native 
Setting up Steam content in /home/japhir/.local/share/Steam
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
/home/japhir/.local/share/Steam/steam.sh: line 99: no match: **:
/home/japhir/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
Running Steam on arch  64-bit
STEAM_RUNTIME is disabled by the user
Installing breakpad exception handler for appid(steam)/version(0)
*** Error in `/home/japhir/.local/share/Steam/ubuntu12_32/steam': free(): invalid pointer: 0xf777c530 ***
ILocalize::AddFile() failed to load file "public/steambootstrapper_english.txt".
[2017-09-26 07:47:30] Startup - updater built Nov 23 2016 01:05:42
crash_20170926074730_1.dmp[2119]: Uploading dump (out-of-process)
/tmp/dumps/crash_20170926074730_1.dmp
/home/japhir/.local/share/Steam/steam.sh: line 713:  2116 Aborted                 $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
/home/japhir/.local/share/Steam/steam.sh: line 444: no match: ssfn*
crash_20170926074730_1.dmp[2119]: Finished uploading minidump (out-of-process): success = yes
crash_20170926074730_1.dmp[2119]: response: CrashID=bp-a79ab0c8-5bc2-421b-82c2-98f742170925
crash_20170926074730_1.dmp[2119]: file ''/tmp/dumps/crash_20170926074730_1.dmp'', upload yes: ''CrashID=bp-a79ab0c8-5bc2-421b-82c2-98f742170925''

**Comment by japhir** [comment]
I just realized I already made an issue for this a long time ago that never got resolved #4945 ;-). Looks like it's still the same?

**Comment by kisak-valve** [comment]
Looks like it. Closing in favor of the #4945.
